src/index.js

```diff
@@ -130,109 +265,113 @@ export const mapProps = mapper => func => {
   return next
 }
 
-export const variant = ({ key, prop = 'variant' }) => {
-  const fn = props => get(props.theme, [key, props[prop]].join('.'), null)
-  fn.propTypes = {
-    [prop]: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-  }
-  return fn
-}
+// Variant
+export const variant = ({ key, prop = 'variant' }) => props =>
+  get(props.theme, [key, props[prop]].join('.')) || null
 
-// space
-const spaceScale = [0, 4, 8, 16, 32, 64, 128, 256, 512]
+// Units
+export const px = value => (num(value) && value !== 0 ? `${value}px` : value)
+export const getPx = px
+export const getWidth = n => (!num(n) || n > 1 ? px(n) : `${n * 100}%`)
 
-const getSpace = (n, scale) => {
-  if (!num(n)) {
-    return px(get(scale, n, n))
-  }
+// Scale
+const DEFAULT_SPACE = [0, 4, 8, 16, 32, 64, 128, 256, 512]
 
-  const isNegative = n < 0
-  const absolute = Math.abs(n)
-  const value = get(scale, absolute)
+const scale = (defaultVariants = DEFAULT_SPACE) => (
+  transformedValue,
+  { rawValue, variants = defaultVariants }
+) => {
+  if (!num(rawValue)) {
+    return px(variants[rawValue] || rawValue)
+  }
+  const abs = Math.abs(rawValue)
+  const neg = negative(rawValue)
+  const value = variants[abs] || abs
   if (!num(value)) {
-    return isNegative ? '-' + value : value
+    return neg ? `-${value}` : value
   }
-  return px(value * (isNegative ? -1 : 1))
+  return px(value * (neg ? -1 : 1))
 }
 
+// Space
 export const margin = style({
   prop: 'margin',
   alias: 'm',
+  cssProperties: ['margin'],
   key: 'space',
-  transformValue: getSpace,
-  scale: spaceScale,
+  transformValue: scale(),
```

Comment:
Why is this called here?

Reply:
The `transformValue` function is now much more permissive than before, so we can even implement the all `space` logic in it.

`scale()` is a transform function factory that gives this possibility. It is a factory because of `fontSize` that has a custom `scale`.